src/core/Akka.Remote/RemoteActorRefProvider.cs

```diff
@@ -288,7 +290,13 @@ internal IInternalActorRef ResolveActorRefWithLocalAddress(string path, Address
             {
                 //the actor's local address was already included in the ActorPath
                 if (HasAddress(actorPath.Address))
-                    return (IInternalActorRef)ResolveActorRef(actorPath);
+                {
+                    // HACK: needed to make ActorSelections work
+                    if (actorPath.ToStringWithoutAddress().Equals("/"))
+                        return RootGuardian;
+                    return _local.ResolveActorRef(RootGuardian, actorPath.Elements);
```

Comment:
previous code wouldn't correctly resolve `DeadLetters` before when resolving a non-existent actor, thus any `Watch` that came after an actor had already died would never be correctly acked. This fixes that.

HOWEVER, using the code as-is in Scala resulted in all `ActorSelection`s not being able to be delivered. Hence why I had to add:

```
// HACK: needed to make ActorSelections work
 +                    if (actorPath.ToStringWithoutAddress().Equals("/"))
 +                        return RootGuardian;
```

That's a special case for `ActorSelection`. Not sure what the "correct" implementation looks like but that's still better than what we had before.